### ISSUE TITLE
Add "favorite" button to traces

### ISSUE THREAD

**Issue by skull8888888** [lmnr-ai/lmnr#173]
Add star icon to the trace row to mark this trace as favorite. Similar to how to can "star" emails in the gmail client.
![Screenshot 2024-11-06 at 10 57 36 PM](https://github.com/user-attachments/assets/c93fd715-6840-4d14-a7fd-5933959707f9)

**Comment by patricksolka** [comment]
Hey @skull8888888 is this issue only related to the UX, meaning that the rest of the functionality is already implemented in the backend or does this also needs to be done with this issue?

**Comment by skull8888888** [comment]
@patricksolka the entire thing needs to be implemented, but I'm happy to make a DB design and API routes and collaborate on this issue together. Did you want to add the UI?

**Comment by patricksolka** [comment]
@skull8888888 Yes, I'd love to work on the UI and contribute to this repo!

**Comment by skull8888888** [comment]
@patricksolka awesome, you can go ahead and start with frontend, just add star icon from lucide-icons as another column and add "add" and "remove" functions placeholders.

**Comment by patricksolka** [comment]
Hey @skull8888888, I followed the steps in Contributing.md to get the frontend up and running. The DB connection seems fine, but I'm getting this error: 

`PostgresError: constraint "labelling_queue_data_queue_id_fkey" of relation "labeling_queue_items" does not exist`

Just to mention, I haven't made an changes to the .env yet.

Could you help me out with this? 

Best wishes! ✌🏻

**Comment by dinmukhamedm** [comment]
Hey @patricksolka sorry about this. Yeah, there were a couple problems with migrations, as we had to edit some of them manually when resolving conflicts. I did merge a change yesterday that should have removed this issue. Could you please try and sync the latest `dev` or `main` branch and try again? Let us know if this doesn't help.

**Comment by dinmukhamedm** [comment]
@patricksolka yes, you can use `observe` with any functions. For example:

```python
from lmnr import Laminar, observe
import time

# Laminar.initialize(...)

@observe
def foo(greeting: str = "Hello"):
    time.sleep(1)
    return f"{greeting} world"

foo()
```

or

```javascript
import { Laminar, observe} from '@lmnr-ai/lmnr'

// Laminar.intialize(...)

const foo = async greeting => `${greeting} world`;

observe(
  { name: 'foo' },
  foo,
  'Hello'
)
.then((out) => {
  console.log(out);
  Laminar.shutdown();
})
```

**Comment by patricksolka** [comment]
Hey @skull8888888 just pushed a PR

**Comment by skull8888888** [comment]
@patricksolka awesome! Will look deeper into during the weekend, from first glance looks good.

**Comment by patricksolka** [comment]
@skull8888888 yeah sure, just let me know if there's something you want me to change!
